Sources/NIOExtras/PCAPRingBuffer.swift

```diff
@@ -12,18 +12,21 @@
 //
 //===----------------------------------------------------------------------===//
 
+import NIOConcurrencyHelpers
 import NIOCore
 
 // MARK: NIOPCAPRingBuffer
 /// Storage for the most recent set of packets captured subject to constraints.
 /// Use ``addFragment(_:)`` as the sink to a ``NIOWritePCAPHandler`` and call ``emitPCAP()``
 /// when you wish to get the recorded data.
-/// - Warning:  This class is not thread safe so should only be called from one thread.
-public class NIOPCAPRingBuffer {
-    private var pcapFragments: CircularBuffer<ByteBuffer>
-    private var pcapCurrentBytes: Int
+/// - Note: This class is annotated `@unchecked Sendable` because its mutable state is protected by a lock.
+public final class NIOPCAPRingBuffer: @unchecked Sendable {
     private let maximumFragments: Int
     private let maximumBytes: Int
+    
+    private let pcapLock = NIOLock()
+    private var pcapFragments: CircularBuffer<ByteBuffer>
+    private var pcapCurrentBytes: Int
```

Comment:
Rather than using a lock should we create a `State` struct and stick that in a locked-value-box?

Reply:
Yes, that would be better